src/sql/workbench/contrib/notebook/browser/notebook.component.ts

```diff
@@ -307,6 +302,32 @@ export class NotebookComponent extends AngularDisposable implements OnInit, OnDe
 		this.model.onScroll.fire();
 	}
 
+	public clickOffCell(e: MouseEvent) {
+		if (this.model.activeCell?.isEditMode) {
+			// when clicking outside of cells toggle edit mode to false
+			this.toggleEditMode();
+		} else {
+			this.unselectActiveCell();
+		}
+	}
+
+	public clickOnCell(cell: ICellModel, event?: MouseEvent) {
+		if (event) {
+			event.stopPropagation();
+		}
+		if (!this.model.activeCell || this.model.activeCell.id !== cell.id) {
+			if (this.model.activeCell?.isEditMode) {
+				// before activating the new cell, change the edit mode of the current active cell
+				this.toggleEditMode();
```

Comment:
It really feels like we should have an editMode setter...all this checking for isEditMode and then calling toggle seems like a recipe for a bug later on. And in most of these cases we don't actually want to toggle it - we want to specifically set it to on or off.